tests/test_bookmarks_cursor.py

```diff
@@ -0,0 +1,128 @@
+import os
+
+import singer
+
+import tap_tester.connections as connections
+import tap_tester.menagerie   as menagerie
+import tap_tester.runner      as runner
+
+from base import TestSquareBaseParent
+
+LOGGER = singer.get_logger()
+
+
+class TestSquareIncrementalReplicationCursor(TestSquareBaseParent.TestSquareBase):
+
+    @staticmethod
+    def name():
+        return "tap_tester_square_incremental_replication_cursor"
+
+    def testable_streams_dynamic(self):
+        return self.dynamic_data_streams().intersection(
+            self.expected_full_table_streams()).difference(
+                self.untestable_streams())
+
+    @staticmethod
+    def testable_streams_static():
+        """ No static streams contain cursor bookmarks. """
+        return set()
+
+    @classmethod
+    def tearDownClass(cls):
+        print("\n\nTEST TEARDOWN\n\n")
+
+    def test_run(self):
+        """Instantiate start date according to the desired data set and run the test"""
+
+        self.START_DATE = self.get_properties().get('start_date')
+
+        print("\n\nTESTING WITH DYNAMIC DATA IN SQUARE_ENVIRONMENT: {}".format(os.getenv('TAP_SQUARE_ENVIRONMENT')))
+        self.bookmarks_test(self.testable_streams_dynamic().intersection(self.sandbox_streams()))
+
+        self.set_environment(self.PRODUCTION)
+        production_testable_streams = self.testable_streams_dynamic().intersection(self.production_streams())
+        if production_testable_streams:
```

Comment:
I don't like this pattern of setting expectations  and running a test using conditional logic. I think we should know if that intersection returns a set and run the test accordingly.

Reply:
I added this because sometimes I want to only run the test for one stream, but if I do that both tests still run which is weird.

Reply:
Going to leave this pattern in, let me know if it's a problem

Reply:
If it makes it easier to work with then I think it's fine.